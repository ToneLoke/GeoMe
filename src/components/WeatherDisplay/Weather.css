.weather-display-container{
  width: 100%;
  height: 100% ;
  background-size: auto ;
  background-repeat: no-repeat;
  color: white;
  }
