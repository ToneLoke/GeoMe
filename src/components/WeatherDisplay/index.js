import React, {Component} from 'react'
import { Col } from 'antd'
import moment from 'moment'
import {GiphyAPI} from '../../apiAdapter'
import './Weather.css'

class FiveDay extends Component {
  state={
    gif: ''
  }
  componentWillMount () {
    // this.gifChange(this.props.day.main)
  }
  componentWillReceiveProps (nextProps) {
    this.gifChange(nextProps.day.main)
  }
  shouldComponentUpdate(nextProps, nextState){
    return Object.is(this.props.day, nextProps.day)  ? true : false
  }
  gifChange = (text) => {
    GiphyAPI.getRandGif(text)
      .then( gif => {
        window.screen.width < 800 ? this.setState({gif: gif.data.fixed_height_downsampled_url}) : this.setState({gif: gif.data.image_original_url})

      })
  }
  render () {
    let {day, updateTime} = this.props
    let d = moment(day.dt_txt)
    return (
      <div style={{backgroundImage: `url(${this.state.gif})`}} className='weather-display-container'>
        <h1>{d.format('MMM DD')}</h1>
        <img src={`http://openweathermap.org/img/w/${day.icon}.png`} />
        <h1 >{Math.floor(day.temp)}<sup>ËšF</sup></h1>
        <h1>{day.description}</h1>
      </div>
    )
  }
}

export default FiveDay
