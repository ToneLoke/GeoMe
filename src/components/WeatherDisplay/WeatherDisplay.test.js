import React from 'react'
import ReactDOM from 'react-dom'
import WeatherDisplay from './index'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

test('Weather display renders the appropiate props', () => {
  const day = { description: 'clear sky', dt_txt: '2017-11-23 00:00:00', icon: '01n', main: 'Clear', temp: 40.73}
  const rendered = renderer.create(
    <WeatherDisplay day={day} />
  )
  expect(rendered.toJSON()).toMatchSnapshot()
})
