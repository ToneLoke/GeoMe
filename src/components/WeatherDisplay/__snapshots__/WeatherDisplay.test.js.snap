// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather display renders the appropiate props 1`] = `
<div
  className="weather-display-container"
  style={
    Object {
      "backgroundImage": "url()",
    }
  }
>
  <h1>
    Nov 23
  </h1>
  <img
    src="http://openweathermap.org/img/w/01n.png"
  />
  <h1>
    40
    <sup>
      ËšF
    </sup>
  </h1>
  <h1>
    clear sky
  </h1>
</div>
`;
